File: app/src/main/java/com/example/pam_app/presenter/IncomePresenter.java
package com.example.pam_app.presenter;

import com.example.pam_app.view.IncomeView;

import java.lang.ref.WeakReference;

public class IncomePresenter {

    private final WeakReference<IncomeView> incomeView;

    public IncomePresenter(final IncomeView incomeView) {
        this.incomeView = new WeakReference<>(incomeView);
    }

    public void onIncomeLeftAmountReceived(final Double incomeLeft) {
        if (incomeView.get() != null) {
            incomeView.get().setUpIncomeLeftText(incomeLeft > 0);
        }
    }
}

File: app/src/main/java/com/example/pam_app/presenter/IncomePresenter.kt
package com.example.pam_app.presenter

import com.example.pam_app.view.IncomeView
import java.lang.ref.WeakReference
import kotlin.jvm.Synchronized

class IncomePresenter(incomeView: IncomeView?) {
    private val incomeView: WeakReference<IncomeView?>
    fun onIncomeLeftAmountReceived(incomeLeft: Double) {
        if (incomeView.get() != null) {
            incomeView.get()!!.setUpIncomeLeftText(incomeLeft > 0)
        }
    }

    init {
        this.incomeView = WeakReference(incomeView)
    }
}